docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc

```diff
@@ -0,0 +1,249 @@
+
+[source,console]
+--------------------------------------------------
+POST api/security_ai_assistant/prompts/_bulk_action
+{
+   "create":[
+      {
+         "name":"test",
+         "id":"test",
+         "content":"some test content",
+         "color":"#D36086",
+         "categories":[],
+         "promptType":"quick",
+         "consumer":"securitySolutionUI"
+      }
+   ]
+}
+--------------------------------------------------
+
+[discrete]
+===== Response code
+
+`200`::
+    Indicates a successful call.
+
+[discrete]
+===== Response payload
+
+JSON object containing the action's outcome:
+
+- `attributes.summary.total`: Total number of prompts matching the bulk action
+- `attributes.summary.succeeded`: Number of successful outcomes (number of prompts that were created, deleted, or updated)
+- `attributes.summary.failed`: Number of failed outcomes
+- `attributes.summary.skipped`: Number of prompts that were skipped due to various reasons (explained below)
+- `attributes.results.created`: Prompt objects that were created during the action's execution
+- `attributes.results.updated`: Prompt objects that were updated during the action's execution
+- `attributes.results.deleted`: Prompt objects that were deleted during the action's execution
+- `attributes.results.skipped`: Prompt that were skipped during the action's execution
+
+A prompt can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a filed to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include prompts' `id`, `name`, and `skip_reason`.
+
+[source,json]
+--------------------------------------------------
+{
+    "success": true,
+    "prompts_count": 1,
+    "attributes": {
+        "results": {
+            "updated": [],
+            "created": [
+                {
+                    "timestamp": "2024-08-13T20:24:08.610Z",
+                    "users": [
+                        {
+                            "id": "testuser",
+                            "name": "elastic"
+                        }
+                    ],
+                    "content": "dasdas",
+                    "updatedAt": "2024-08-13T20:24:08.610Z",
+                    "id": "0B1pTZEBYaDeA-NhjHej",
+                    "name": "test",
+                    "promptType": "quick",
+                    "color": "#D36086",
+                    "categories": [],
+                    "consumer": "securitySolutionUI"
+                }
+            ],
+            "deleted": [],
+            "skipped": []
+        },
+        "summary": {
+            "failed": 0,
+            "succeeded": 1,
+            "skipped": 0,
+            "total": 1
+        }
+    }
+}
+--------------------------------------------------
+
+*Example 2, Partial failure*
+
+The following request adds tags `tag-1` and `tag-2` to the rules that have the IDs sent in the payload:
```

Comment:
```suggestion
The following request adds tags `tag-1` and `tag-2` to the rules associated with the IDs sent in the payload:
```

Reply:
The request doesn't look like it has `tag-1` or `tag-2` in it. I could be missing something but how would it add those tags without mentioning them? And I thought this page was about prompts, not rules?